### PR TITLE
HTTP Cleanups

### PR DESCRIPTION
**Pull request by doanac**
This consolidates some http handling into a common file. It's just some generic cleanup logic that will allow the cert rotation PR to have less code churn.

### CODE REVIEW

File: internal/config.go [hunk 1]
```diff
@@ -44,3 +48,60 @@ func UnmarshallBuffer(c CryptoHandler, encContent []byte, decrypt bool) (ConfigS
...
+	Unencrypted bool     `json:"unencrypted"`
+	OnChanged   []string `json:"on-changed,omitempty"`
+}
+
+type ConfigCreateRequest struct {
+	Reason string          `json:"reason"`
+	Files  []ConfigFileReq `json:"files"`
+}
+
+func updateConfig(app *App, client *http.Client, pubkey string) error {
```
Comment (vkhoroz) [inline internal/config.go:64]: As I understand now this function is unused without the ifdef, making the linter unhappy.

This change is extremely trivial though; so I recommend to take it wherever you planned to reuse this function (as a separate PR commit).

**Review by vkhoroz — APPROVED** [review]
LGTM

**Review by vkhoroz — APPROVED** [review]

File: internal/config.go (hunk 1 above)
Comment (doanac) [reply internal/config.go:64]: I'm trying a slightly different approach. I've changed our linter to include vpn.go which found a bug.

I think doing it all here is going to make your life easier when you finally see the cert-rotation PR

Comment (doanac) [reply internal/config.go:64]: Take a look at my force push and make sure you like it.

**Pull request merged**